### PR TITLE
FEATURE: Allow group moderators to close/archive topics

### PR DESCRIPTION
**Pull request by jbrw**
Rename enable_category_group_review to enable_category_group_moderation

### CODE REVIEW

File: app/assets/javascripts/discourse/app/widgets/topic-admin-menu.js [hunk 1]
```diff
@@ -158,7 +158,12 @@ export default createWidget("topic-admin-menu", {
           label: "actions.recover"
         });
       }
+    }
 
+    if (
+      this.currentUser &&
+      (this.currentUser.get("canManageTopic") || details.get("can_close_topic"))
```
Comment (eviltrout) [inline app/assets/javascripts/discourse/app/widgets/topic-admin-menu.js:165]: Do we still need the user and `canManageTopic` checks here if we can check `can_close`, `can_archive`, etc?

File: config/locales/server.be.yml [hunk 2]
```diff
@@ -1239,7 +1239,7 @@ be:
     staff_user_custom_fields: "Спіс карыстальнікаў карыстацкіх палёў, якія могуць быць атрыманы для супрацоўнікаў з API."
     enable_user_directory: "Пакажыце каталог карыстальнікаў для прагляду"
     enable_group_directory: "Забяспечыць каталог груп для прагляду"
-    enable_category_group_review: "Дазволіць групы для разгляду ўтрымання ў канкрэтных катэгорыях"
```
Comment (eviltrout) [inline config/locales/server.be.yml:1242]: We needn't check in these things for other languages - they will be automatically flagged in transifex when the keys change.

File: lib/guardian/topic_guardian.rb [hunk 3]
```diff
@@ -199,4 +199,18 @@ def can_edit_tags?(topic)
 
     false
   end
+
+  def can_perform_action_available_to_group_moderators?(topic)
+    return false if anonymous? || topic.nil?
+    return true if is_staff?
+    return true if @user.has_trust_level?(TrustLevel[4])
```
Comment (eviltrout) [inline lib/guardian/topic_guardian.rb:206]: Could trust level 4 previously close and archive topics? I didn't think they could!

Comment (jbrw) [reply lib/guardian/topic_guardian.rb:206]: I _think_ so. Please double check the following for me.

canManageTopic is true for isStaff or isElder, where isElder is someone with a trust level of 4:

https://github.com/discourse/discourse/blob/33554e5cbc58463190a8257eb3bd8875b4c3479f/app/assets/javascripts/discourse/app/models/user.js#L218

Having canManageTopic exposes many admin options to someone with a trust level of 4:

https://github.com/discourse/discourse/blob/0d6b638dac161980ee39631271fd5b3f00fe9b14/app/assets/javascripts/discourse/app/widgets/topic-admin-menu.js#L133

Many of these actions (ie, pinned pinned_globally visible closed archived) end up hitting TopicsController#status, with the work happening around lines 413-416:

https://github.com/discourse/discourse/blob/0d6b638dac161980ee39631271fd5b3f00fe9b14/app/controllers/topics_controller.rb#L413-L416

which ends up calling the following:

https://github.com/discourse/discourse/blob/33554e5cbc58463190a8257eb3bd8875b4c3479f/lib/guardian.rb#L163-L165

where trust level 4 is checked against.

File: app/assets/javascripts/discourse/app/widgets/topic-admin-menu.js (hunk 1 above)
Comment (jbrw) [reply app/assets/javascripts/discourse/app/widgets/topic-admin-menu.js:165]: You're right, canManageTopic is redundant when used alongside can_close_topic or can_archive_topic. I will remove them.

File: lib/guardian/topic_guardian.rb (hunk 3 above)
Comment (eviltrout) [reply lib/guardian/topic_guardian.rb:206]: Well that is very convincing evidence! Thanks for the leg work.

**Pull request merged**